servicetest/testframework/lib.py

```diff
@@ -380,14 +380,22 @@ def __init__(self,
                  config_path=None,
                  caps_dir=None,
                  default_caps_dir=None,
-                 check_connection=True):
+                 check_connection=True,
+                 exec_prefix=None):
         if log_file_path is None:
             self.__log_file = subprocess.STDOUT
         else:
-            self.__log_file = open(log_file_path, "w")
+            if os.path.exists(log_file_path):
+                os.unlink(log_file_path)
```

Comment:
It depends on what one means with old runs. We create a new agent for each test class, so if we have more than one test class in the same module, we only get the logs from the last one.